### PR TITLE
feat:✨ Data table is added for orca

### PR DESCRIPTION
**Pull request by jyothiprakashk**

### CODE REVIEW

File: web/src/datatable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import React, { useState } from "react";
+import { IColumnItems } from "../interface/datatable";
+import { DataModal } from "./modal";
+import { Button, notification } from "antd";
+import styles from "./datatable.module.css";
+import { Input } from "antd";
```
Comment (rakeshar3796) [inline web/src/datatable/index.tsx:6]: @jyothiprakash-frontend rearrange this

File: web/src/datatable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      setColumns([...columns, item]);
+      handleCancel();
+      setColumnName("");
+    }
+    return;
+  };
+
+  const updateState = (e: any) => {
+    let target = e.target.attributes as any;
+    console.log(target, "updateState");
```
Comment (rakeshar3796) [inline web/src/datatable/index.tsx:43]: remove consoles

File: web/src/datatable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    let fieldValue = e.target.value;
+    const tempObj = rows[index] as any;
+    tempObj[prope] = fieldValue;
+    rows[index] = tempObj;
+    setRows(rows);
+  };
+
+  const postResults = () => {
+    let isToast = false;
+    rows.map((each) => Object.keys(each).length >= 1)
```
Comment (rakeshar3796) [inline web/src/datatable/index.tsx:55]: what is this da?

File: web/src/datatable/index.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  };
+
+  const handleOnSubmit = () => {
+    handleAddColumn();
+  };
+
+  return (
+    <div>
+      <div className="container">
+        <div>
```
Comment (rakeshar3796) [inline web/src/datatable/index.tsx:91]: do we need this much divs?

File: web/src/datatable/index.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                  {rows.map((item, idx) => (
+                    <tr key={idx}>
+                      <td>{idx + 1}</td>
+                      {columns.map((column, index) => (
+                        <td key={index}>
+                          <Input
+                            type="text"
+                            data-column={column.name}
+                            data-index={idx}
+                            value={rows?.idx?.column}
```
Comment (rakeshar3796) [inline web/src/datatable/index.tsx:117]: remove null check for rows

File: web/src/datatable/index.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+          <Button onClick={handleAddRow} type="primary">
+            +
+          </Button>
+        </div>
+        <Button
+          onClick={postResults}
+          type="primary"
+          className={styles.saveButton}
+        >
+          save
```
Comment (rakeshar3796) [inline web/src/datatable/index.tsx:148]: Save

File: web/src/datatable/modal.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
+import { Modal, Button, Input } from "antd";
+import style from "./datatable.module.css";
```
Comment (rakeshar3796) [inline web/src/datatable/modal.tsx:2]: add linebreak

File: web/src/datatable/index.tsx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+  const showModal = () => {
+    setColumnName("");
+    setIsModalVisible(true);
+    setColumnError("");
+  };
+
+  const handleAddRow = () => {
+    let empty = {};
+    setRows([...rows, empty]);
```
Comment (gopal1996) [inline web/src/datatable/index.tsx:25]: this will create like [...., {}].. expected?

File: web/src/datatable/index.tsx [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    tempObj[prope] = fieldValue;
+    rows[index] = tempObj;
+    setRows(rows);
+  };
+
+  const postResults = () => {
+    let isToast = false;
+    rows.map((each) => Object.keys(each).length >= 1)
+      ? console.log(rows)
+      : (isToast = true);
```
Comment (gopal1996) [inline web/src/datatable/index.tsx:57]: you can improve the logic

File: web/src/datatable/index.tsx [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+                  handleCancel={handleCancel}
+                  columnName={columnName}
+                  columnError={columnError}
+                />
+              }
+            </div>
+          </div>
+          <Button onClick={handleAddRow} type="primary">
+            +
+          </Button>
```
Comment (gopal1996) [inline web/src/datatable/index.tsx:141]: font size will be small crct?

File: web/src/datatable/modal.tsx [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+interface ModalProps {
+  handleColumnChange: any;
+  handleOnSubmit: any;
+  showModal: any;
+  isModalVisible: any;
+  handleCancel: any;
+  columnName: string;
+  columnError: string;
+}
```
Comment (gopal1996) [inline web/src/datatable/modal.tsx:11]: why any type ?

File: web/src/datatable/modal.tsx [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+      </Button>,
+      <Button
+        type="primary"
+        onClick={handleOnSubmit}
+        disabled={columnError.length >= 1}
+      >
+        Okay
+      </Button>
+    ];
+  };
```
Comment (gopal1996) [inline web/src/datatable/modal.tsx:37]: wrap this with memo

File: web/src/datatable/index.tsx (hunk 10 above)
Comment (jyothiprakash-frontend) [reply web/src/datatable/index.tsx]: you mean + button

File: web/src/datatable/modal.tsx (hunk 12 above)
Comment (jyothiprakash-frontend) [reply web/src/datatable/modal.tsx]: why memo here,no re-renders right?

**Review by rakeshar3796 — APPROVED** [review]

**Pull request merged**